src/worker/app.ts

```diff
@@ -361,8 +374,33 @@ export class App {
     });
     logger.debug("Done.");
 
-    setTimeout(app.run, interval);
+    if (this.runTimeout) {
+      clearTimeout(this.runTimeout);
+      this.runTimeout = undefined;
+    }
+
+    this.runTimeout = setTimeout(app.run, interval);
 
     return Promise.resolve(true);
   }
+
+  public stop(): void {
+    if (this.runTimeout) {
+      clearTimeout(this.runTimeout);
+      this.runTimeout = undefined;
+    }
+
+    if (this.switchbot) {
+      this.switchbot.stopScan();
+      this.switchbot.onadvertisement = undefined;
+    }
+
+    if (this.socket) {
+      this.socket.close();
+      this.socket = undefined;
+    }
+
+    this.initialized = false;
+    App._instance = undefined;
+  }
```

Comment:
Added this so tests can properly start and stop workers without leaving behind open handles.